[WIP] add helicopter page (#2216)
PX4/PX4-user_guide: en/config_heli/README.md

### Rate Controller
The rate controller should be tuned in Acro, but can also be done in Stabilized if you cannot fly Acro.

Start off with disabled rate controller gains, and only some feedforward:
```
param set MC_ROLLRATE_P 0
param set MC_ROLLRATE_I 0
param set MC_ROLLRATE_D 0
param set MC_ROLLRATE_FF 0.1
param set MC_PITCHRATE_P 0
param set MC_PITCHRATE_I 0
param set MC_PITCHRATE_D 0
param set MC_PITCHRATE_FF 0.1
```
Take off slowly and provide some roll and stick movements. Use the QGC tuning ui to check the response:
![QGC Rate Controller Tuning UI](../../assets/mc_pid_tuning/qgc_mc_pid_tuning_rate_controller.png)
Increase the FF gains until the response reaches the setpoint when giving a step input.

Then enable the PID gains. Start off with P=FF/4, I=0.2 and D=0.001. Then increase the P and D gains as needed until it tracks well. It is expected that the P gain is considerably smaller than the FF gain.

### Yaw Compensation
There are two parameters to compensate yaw for the main rotor's collective and throttle.
A negative value is needed when positive thrust of the tail rotor rotates the vehicle opposite to the main rotor turn direction.

> TODO: add log + explain

Author: @MaEtUgR do you still have a good log here?

Reviewer: @MaEtUgR ^^^

Reviewer: @MaEtUgR Do we have a log. If he doesn't @bkueng What do we do?

# Helicopter Configuration

This section contains topics related to helicopter configuration and tuning.

:::note

> Helicopters are less well-supported than other types of vehicles.

Reviewer: What do we mean by well-supported here? I mean I presume a helicopter supports all the same modes as multicopter in the same way - including offboard?

What other types of frames are there?
What features might be missing?

Author: Modes yes.
In terms of frames, there's e.g. coaxial or dual rotor types. Features include RPM governor or autorotation.

We would welcome your [contribution](../contribute/README.md) of new features, new frame configurations, or other improvements.
:::

## Supported Configurations

> Supported helicopter configurations:
> - Single main rotor with any type of swash-plate
> - ESC tail rotor

Reviewer: This reads as two configurations - a single main rotor and possibly a single main rotor + esc tail rotor: [Generic Helicopter (Tail ESC)](https://docs.px4.io/main/en/airframes/airframe_reference.html#copter_helicopter_generic_helicopter_(tail_esc)).

Given you can't indicate either of these are missing, do you mean:

```suggestion
Supported helicopter configurations:
- Single main rotor with any type of swash-plate and an ESC tail rotor.
```

> - ESC tail rotor

Reviewer: What is an ESC tail rotor? i.e. given there is no configuration of this rotor does this just mean you attach an ESC + motor to the tail and everything the system will magically work out everything else it needs to do?

## Setup

To setup and configure a helicopter: 

1. Select the helicopter [Airframe](../config/airframe.md) in QGroundControl
1. Configure [Actuator](../config/actuators.md) geometry

>    ![Geometry helicopter](../../assets/config/actuators/qgc_geometry_helicopter.png)

Reviewer: Would you happen to have a complete configuration screen for the helicopter? ("as well")?

Reviewer: @bkueng If you have a setup screen for the whole helicopter actuator config that would be good (i.e. for a real build). I would use it for a separate helicopter config page, but below the current actuator config. In a further PR.

   1. Remove the rotor blades and propellers
   1. Configure the swash-plate servos.
      The angle is measured clockwise with 0 pointing forwards.
      Trim is typically not needed, but can be used to offset individual servo positions.

>    1. Set the main rotor turning orientation

Reviewer: I guess options are disabled/enabled?

Author: As you copied these over to the actuator page, did you still want to keep this here?

Reviewer: Good point. Removed this bit.

1. Assign motors and servos to outputs and test (also in [Actuator configuration](../config/actuators.md))
   1. Assign the motors and servos to the outputs
   1. Power the vehicle with a battery and use the actuator testing sliders to validate correct servo and motor assignment and direction.
1. Using an RC in [Acro mode](../flight_modes/acro_mc.md), verify the correct movement of the swash-plate:
   - moving the Roll stick to the right should tilt the swash-plate to the right.
   - moving the Pitch stick forward should tilt the swash-plate forward.
1. Arm the vehicle and check the main rotor spins up slowly.

>    Adjust the throttle spoolup time as needed.

Reviewer: What criteria?

   You can also adjust the throttle curve with the parameters [CA_HELI_THR_Cx](../advanced_config/parameter_reference.md#CA_HELI_THR_C0).

>    The default is constant, maximum throttle.

Reviewer: I presume that the throttle curve adjusts the throttle non-linearly across the RC stick position for some reason. Is there something we can/should link here to make this clear, or to advise what makes sense for most users?

Author: Yes. The default makes sense for most. I don't have enough experience here either.

3. Disarm again and power off.
4. Put the rotor blades on and power the vehicle.
5. Configure the collective pitch curve using the parameters [CA_HELI_PITCH_Cx](../advanced_config/parameter_reference.md#CA_HELI_PITCH_C0).

>   Set the minimum and maximum according to the minimum and maximum blade angles you want.

Reviewer: There is a whole lot of terminology here that is really only familiar if you work with helicopters - e.,g. collective etc. 

I assume that setting these minimum and maximum and all the rest of this is obvious to someone who understands helicopters (I have no idea what blade angles I might want)?

Author: Yes :)

  Make sure the minimum is low enough so the vehicle can still descend.
  Instead start off with a too low value.
  The default is slightly negative for that reason and should be a good starting point.

## Tuning

After completing the previous steps you are ready to arm with blades mounted.

> Tuning involves two steps that are specific to helicopters, the rest is the same as for multicopters.

Reviewer: 1. Can I autotune.
2. What does "the rest" mean? I.e. what specific additional instructions can we link to that users should follow?

Author: 1. No. Or at least I would not try to.
2. The MC tuning guides (specifically for attitude, velocity and position controllers)

### Rate Controller

The rate controller should be tuned in [Acro mode](../flight_modes/acro_mc.md), but can also be done in [Stabilized mode](../flight_modes/manual_stabilized_mc.md) if you cannot fly Acro mode.

Start off with disabled rate controller gains, and only some feedforward:

```
param set MC_ROLLRATE_P 0
param set MC_ROLLRATE_I 0
param set MC_ROLLRATE_D 0
param set MC_ROLLRATE_FF 0.1
param set MC_PITCHRATE_P 0
param set MC_PITCHRATE_I 0
param set MC_PITCHRATE_D 0
param set MC_PITCHRATE_FF 0.1
```

Take off slowly and provide some roll and stick movements.
Use the QGC tuning UI to check the response:

![QGC Rate Controller Tuning UI](../../assets/mc_pid_tuning/qgc_mc_pid_tuning_rate_controller.png)

Increase the FF gains until the response reaches the setpoint when giving a step input.

Then enable the PID gains.
Start off with `P=FF/4`, `I=0.2` and `D=0.001`.
Then increase the `P` and `D` gains as needed until it tracks well.
It is expected that the `P` gain is considerably smaller than the `FF` gain.

### Yaw Compensation

> There are two parameters to compensate yaw for the main rotor's collective and throttle.

Reviewer: Which parameters?

# Helicopter Configuration

This section contains topics related to helicopter configuration and tuning.

:::note

> Helicopters are less well-supported than other types of vehicles.
> We would welcome your [contribution](../contribute/README.md) of new features, new frame configurations, or other improvements.

Reviewer: With respect to what this means then https://github.com/PX4/PX4-user_guide/pull/2216/files#r1059837623, how about:

```suggestion
PX4 supports helicopters with coaxial or dual rotor types, and features such as RPM governor and autorotation. It can generally be used for the same kinds of things as other multicopters.

We would welcome your [contribution](../contribute/README.md) of new features, new frame configurations, or other improvements.
```

Reviewer: PS FWIW one day it would be good to have a helicopter introduction page that covers the supported features etc - i.e. that stuff ^^^. That could then be the parent for build logs and so on, in the same way as http://localhost:8080/px4_user_guide/en/frames_vtol/

Author: > PX4 supports helicopters with coaxial or dual rotor types

PX4 *does not support* ...

> Helicopters are less well-supported than other types of vehicles.
> We would welcome your [contribution](../contribute/README.md) of new features, new frame configurations, or other improvements.

Reviewer: ```suggestion
Helicopters are less well-supported on than other types of vehicles.
For example, PX4 does not support helicopters with coaxial or dual rotor types, and features such as RPM governor and autorotation. 
We would welcome your [contribution](../contribute/README.md) of new features, new frame configurations, or other improvements.
```
